diff --git a/util-linux/rtcwake.c b/util-linux/rtcwake.c
index 8aee0cf..6693305 100644
--- a/util-linux/rtcwake.c
+++ b/util-linux/rtcwake.c
@@ -122,7 +122,7 @@ int rtcwake_main(int argc, char **argv) MAIN_EXTERNALLY_VISIBLE;
 int rtcwake_main(int argc UNUSED_PARAM, char **argv)
 {
 	unsigned opt;
-	const char *rtcname = NULL;
+	const char *rtcname = "/dev/rtc0";
 	const char *suspend = "standby";
 	const char *opt_seconds;
 	const char *opt_time;
