--- core/main.c	2013-05-09 15:45:50.000000000 +0200
+++ core/main.c	2013-12-19 14:46:03.020362993 +0100
@@ -1477,9 +1477,9 @@
         /* Set up PATH unless it is already set */
         setenv("PATH",
 #ifdef HAVE_SPLIT_USR
-               "/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
+               "/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin" ":/opt/openslx/sbin:/opt/openslx/bin",
 #else
-               "/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin",
+               "/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin" ":/opt/openslx/sbin:/opt/openslx/bin",
 #endif
                arg_running_as == SYSTEMD_SYSTEM);
 
@@ -1513,6 +1513,11 @@
                 unsetenv("LISTEN_FDS");
                 unsetenv("LISTEN_PID");
 
+                /* OpenSLX: "Fix" libc/nss/nss_ldap deadlock
+                 * when not using nscd */
+                setenv("HOME", "/", 1);
+                setenv("USER", "root", 1);
+
                 /* All other variables are left as is, so that clients
                  * can still read them via /proc/1/environ */
         }
